tests/libwnbd_tests/test_io.cpp

```diff
@@ -644,3 +644,89 @@ TEST(TestPersistentReservations, TestPersistentReserveOut) {
     WnbdGetUserspaceStats(WnbdDisk, &UserspaceStats);
     ASSERT_EQ(UserspaceStats.PersistResOutErrors, 0);
 }
+
+TEST(TestScsiUnmap, TestUnmap) {
+    WNBD_PROPERTIES WnbdProps = { 0 };
+    GetNewWnbdProps(&WnbdProps);
+    MockWnbdDaemon WnbdDaemon(&WnbdProps);
+    WnbdDaemon.Start();
+    WNBD_CONNECTION_INFO ConnectionInfo = { 0 };
+    ASSERT_FALSE(WnbdShow(WnbdProps.InstanceName, &ConnectionInfo))
+        << "couldn't retrieve WNBD disk info";
+
+    std::string DiskPath = GetDiskPath(WnbdProps.InstanceName);
+    PWNBD_DISK WnbdDisk = WnbdDaemon.GetDisk();
+
+    HANDLE DiskHandle = CreateFileA(
+        DiskPath.c_str(),
+        GENERIC_READ | GENERIC_WRITE,
+        FILE_SHARE_READ | FILE_SHARE_WRITE,
+        NULL,
+        OPEN_EXISTING,
+        0,
+        NULL);
+
+    ASSERT_NE(DiskHandle, INVALID_HANDLE_VALUE)
+        << "Error opening device: " << GetLastError();
+
+    std::unique_ptr<void, decltype(&CloseHandle)> DiskHandleCloser(
+        DiskHandle, &CloseHandle);
+
+    UNMAP_BLOCK_DESCRIPTOR UnmapDescriptor = { 0 };
+
+    int LbaCount = 1;
+    REVERSE_BYTES_4(&UnmapDescriptor.LbaCount, &LbaCount);
+    int64_t StartingLba = 0;
+    REVERSE_BYTES_8(&UnmapDescriptor.StartingLba, &StartingLba);
+
+    const int AllocationLength = sizeof(UNMAP_LIST_HEADER) + sizeof(UNMAP_BLOCK_DESCRIPTOR);
+    std::unique_ptr<UNMAP_LIST_HEADER> UnmapListHeader(
+        (PUNMAP_LIST_HEADER) new BYTE[AllocationLength]);
+
+    ZeroMemory(UnmapListHeader.get(), AllocationLength);
+
+    CDB UnmapCdb = { 0 };
+    UnmapCdb.UNMAP.OperationCode = SCSIOP_UNMAP;
+    REVERSE_BYTES_2(&UnmapCdb.UNMAP.AllocationLength, &AllocationLength);
+
+    RtlCopyMemory(UnmapListHeader->Descriptors, &UnmapDescriptor, sizeof(UNMAP_BLOCK_DESCRIPTOR));
```

Comment:
nit: could've filled the first descriptor directly, without using a separate variable.